### DISCUSSION TITLE
Question about DRYCOLOD.mos / weather CSV file

### DISCUSSION THREAD

**Discussion by Norah135** [ibpsa/project1-boptest#668] [Q&A - Test Cases] [answered]
I worked on developing a test case independently and followed the documentation for guidance. However, I have a question regarding the water files: How can I modify the DRYCOLOD.mos file to suit a specific location? Additionally, I’m curious about the creation of the weather CSV file. I understand that it is generated using a simulation, but I’m still unsure about the process. Is it possible to obtain the CSV file without running a simulation model? I went to get the weather csv file in simple way based in the location that I will provide.

**Comment by EttoreZ** [comment]
Dear @Norah135 `.mos` weather files are extensions used by Modelica libraries. However, they are identical in content to `.TMY3` or `.epw` weather files. You can go the E+  [websiste](https://energyplus.net/weather) and get a weather file for your location. Some already have `.mos` available. If that is not the case you can convert your `.epw` following the procedure highlighted here: [weather generation](https://simulationresearch.lbl.gov/modelica/releases/v11.0.0/help/Buildings_BoundaryConditions_WeatherData.html#Buildings.BoundaryConditions.WeatherData.ReaderTMY3), just go to the section "add new weather data". If you have your own `.TMY2 `or` TMY3` weather file, you need to use E+ [converter ](https://bigladdersoftware.com/epx/docs/8-3/auxiliary-programs/using-the-weather-converter.html) to first convert it into an `.epw` and than convert it to `.mos`. The fastest way to generate the `.csv` file is as you mention to simulate jus the [weather reader ](https://simulationresearch.lbl.gov/modelica/releases/v11.0.0/help/Buildings_BoundaryConditions_WeatherData.html#Buildings.BoundaryConditions.WeatherData) and not the whole model (with any modelica tool OpenModelica, Dymola, Optimica), and then adapt this [script](https://github.com/ibpsa/project1-boptest/blob/master/testcases/twozone_apartment_hydronic/models/generate_data.py) to create the `.csv`. If you do not want to use a simulation, you would have to manually creating the '.csv' file.

**Reply by Norah135** [reply]
Thanks for your response @EttoreZ . I have a couple of questions:

Why do I need the .mos file, and what’s the purpose of the simulation process ?
Can I just get the data CSV file by extracting it from a previous file (.mos file) ?

**Reply by EttoreZ** [reply]
The `.mos` file is needed for the simulation of the Modelica model, the `.csv` file is used by BOPTEST to produce the weather forecasts. The simulation is an easy way to generate the `.csv` file starting from the `.mos` file using the script I linked in the previous post. You can manually extract the data from the `.mos` file, but you need to rename it, resample it, and add some variables that are calculated and are not present in the original `.mos` file, but are present in the `.csv` (Altitude Angle,Solar Declination,Solar Hour Angle,Solar Time,Zenith Angle,Black Sky Temperature). These variables are generated with the [weather reader](https://simulationresearch.lbl.gov/modelica/releases/v11.0.0/help/Buildings_BoundaryConditions_WeatherData.html#Buildings.BoundaryConditions.WeatherData) I mentioned before  during the Modelica simulation.  For a detailed explanation on how each Test case is set up you can check the [documentation](https://ibpsa.github.io/project1-boptest/), both user guide and design specification.

**Comment by dhblum** [comment, marked as answer]
Note also that during the simulation of the Modelica model, the weather reader interpolates the .mos file into splines for intervals sub-one hour.  Thus, for BOPTEST to provide accurate deterministic weather forecasts at sub one-hour resolution (if asked for by a user through the API), we wanted to replicate that spline behavior, and therefore run the weather-only simulation to convert the .mos used by the Modelica model into the .csv used by BOPTEST.
